docs/DeveloperGuide.md

```diff
@@ -8,23 +8,119 @@
+
+### Record and RecordList  
+
+The main class in our program is the ```Record``` and ```RecordList``` abstract classes, in which ```Income```, ```Expense``` will inherit from ```Record``` and ```IncomeList``` and ```ExpenseList``` will inherit from ```RecordList```. Most commands will act on instances of the ```Income```, ```Expense```,```IncomeList``` and ```ExpenseList``` classes.  
+
+![Record Class](images/Record_RecordList_UML_class.png)
+
+### DeleteIncomeCommand 
+
+The proposed DeleteIncomeCommand mechanism is facilitated by ```System```, ```Incomes```, ```UI``` and ```ChChingException```.  
+The command receives the instruction from ```UI``` and will call the ```execute``` method.  
+If ```index <= 0```, the command will throw a new ChChingException and print ```"Zero/Negative index"```.  
+If ```index > incomes.size()```, the command will also throw a new ChChingException and print ```"The number is too big"```  
+Entering any of these optional lines will result in early termination of the command.  
+Or else, the command will continue to delete the entry at the particular index.  
+Afterwards, the ```execute()``` method will print ```"Income deleted, here is the updated list:"``` and prints the entries in the income list.
+
+![Record Class](images/DeleteIncomeCommand_sequence_diagram.png)
+
+### Target and TargetStorage
+
+The `Target` and `TargetStorage` class allows users to set a target for their ideal balance.
+
+![Target Class](images/Target_UML.png)
+
+### [Proposed] EditIncomeCommand/EditExpenseCommand
+The proposed edit income command is facilitated by `Parser`, `EditIncomeCommand`, `IncomeList`, while the proposed edit expense command is facilitated by `Parser`, `EditExpenseCommand` and `ExpenseList`. 
+
+Note that below highlights the implementation of the edit income command, with edit expense command following the same implementation.
+
+Given below is how the edit income mechanism works at each step:
+1. The user types in the command for edit income with the necessary arguments given to indicate which income record to edit and what fields to edit for the chosen income record.
+2. The line inputted by the user will then be parsed by `Parser`, where it will check if the input contains a valid index with valid fields to edit. If they are valid, it will return `EditIncomeCommand`. Else it will throw a `ChChingException` indicating an error in the input from the user.
+3. `EditIncomeCommand` will then perform its `execute` method, where for each field to be edited, it will call `editIncome` method from the `ExpenseList` class. This would then update the required income record accordingly.
+
+The following sequence diagram shows how the edit income command works:
+<br> ![edit income sequence diagram](images/EditIncomeCommand_sequence_diagram.png)
+
+The edit expense command works in a similar way, with its sequence diagram as shown:
+<br> ![edit expense sequence diagram](images/EditExpenseCommand_sequence_diagram.png)
```

Comment:
This edit expense sequence diagram may not be necessary due to its similarity to the edit income sequence diagram shown above. On the other hand, this font is clearer and more comprehensive to read as compared to the above sequence diagrams.